src/main/java/org/apache/sysds/runtime/matrix/data/LibMatrixReorg.java

```diff
@@ -389,10 +393,78 @@ public static MatrixBlock rev( MatrixBlock in, MatrixBlock out ) {
 		return out;
 	}
 
+	public static MatrixBlock rev(MatrixBlock in, MatrixBlock out, int k) {
+		if (k <= 1 || in.isEmptyBlock(false)) {
```

Comment:
add a check for a minimum size for parallelization, otherwise fall back to single-threaded as we do it for example in aggregations (LibMatrixAgg).

Reply:
I have added something like 3000_000 in the hop for now, for the tests to trigger the multi-thread.